jobs/kpi-forecasting/kpi-forecasting/models/prophet_forecast.py

```diff
@@ -0,0 +1,119 @@
+import json
+import pandas as pd
+import prophet
+
+from datetime import datetime
+from dataclasses import dataclass
+from models.base_forecast import BaseForecast
+from typing import Dict
+
+
+@dataclass
+class ProphetForecast(BaseForecast):
+    @property
+    def column_names_map(self) -> Dict[str, str]:
+        return {"submission_date": "ds", "value": "y"}
+
+    def _fit(self) -> None:
+        """
+        Fit a Prophet model using the `observed_df` that was generated using
+        Metric Hub. This method updates `self.model`.
+        """
+        self.model = prophet.Prophet(
+            **self.parameters,
+            uncertainty_samples=self.number_of_simulations,
+            mcmc_samples=0,
+        )
+
+        if self.use_holidays:
+            self.model.add_country_holidays(country_name="US")
+
+        # Modify observed data to have column names that Prophet expects, and fit
+        # the model
+        self.model.fit(self.observed_df.rename(columns=self.column_names_map))
+
+    def _predict(self) -> pd.DataFrame:
+        """
+        Forecast using `self.model`.
+        """
+        # generate the forecast samples
+        samples = self.model.predictive_samples(
+            self.dates_to_predict.rename(columns=self.column_names_map)
+        )
+        df = pd.DataFrame(samples["yhat"])
+
+        # add dates_to_predict to the output
+        df["submission_date"] = self.dates_to_predict
+
+        return df
+
+    def _predict_legacy(self) -> pd.DataFrame:
+        """
+        Recreate the legacy format used in
+        `moz-fx-data-shared-prod.telemetry_derived.kpi_automated_forecast_v1`.
+        """
+        # TODO: This method should be removed once the forecasting data model is updated:
+        # https://docs.google.com/document/d/18esfJraogzUf1gbZv25vgXkHigCLefazyvzly9s-1k0.
+
+        df = self.model.predict(
+            self.dates_to_predict.rename(columns=self.column_names_map)
+        )
+
+        # set legacy column values
+        if "dau" in self.metric_hub.alias.lower():
+            df["metric"] = "DAU"
+        else:
+            df["metric"] = self.metric_hub.alias
+
+        df["forecast_date"] = str(datetime.utcnow().date())
+        df["forecast_parameters"] = str(
+            json.dumps({**self.parameters, "holidays": self.use_holidays})
+        )
+
+        if "desktop" in self.metric_hub.alias.lower():
```

Comment:
Is it possible for both `desktop` and `mobile` to be in the set?

Reply:
It is possible but unlikely for now. This code path will be deleted in the near-future when the new data model is complete.

I made the parsing more robust so that it handles the case that you mention.